packages/tools/devtools/devtools-view/src/components/OpLatencyView.tsx

```diff
@@ -44,89 +51,202 @@ const useStyles = makeStyles({
+						data: [...(currentData?.data ?? []), newDataPoint],
+					};
+				});
+				setDurationInboundToProcessingData((currentData) => {
+					const newDataPoint = {
+						timestamp: message.data.event.timestamp,
+						duration: Number(eventContents.durationInboundToProcessing),
+					};
+					return {
+						...currentData,
+						data: [...(currentData?.data ?? []), newDataPoint],
+					};
+				});
+
+				return true;
+			},
+		};
+
+		// Event handler for messages coming from the Message Relay
+		function messageHandler(message: Partial<ISourcedDevtoolsMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers);
+		}
+
+		messageRelay.on("message", messageHandler);
+
+		return (): void => {
+			messageRelay.off("message", messageHandler);
+		};
+	}, [messageRelay]);
+
 	return (
 		<div className={styles.mainContainer} data-testid="test-op-latency-view">
 			<h3>Op Latency</h3>
-			<DynamicComposedChart
-				margin={{
-					top: 15,
-					right: 30,
-					left: -15,
-					bottom: 40,
-				}}
-				legendStyle={{
-					marginLeft: 25,
-					bottom: -5,
-				}}
-				yAxisUnitDisplayName="ms"
-				// NOTE: Because Op Latency data is not yet available, this is a placeholder
-				dataSets={[]}
-			/>
-			<div className={styles.graphAboutContainer}>
-				<div className={styles.flexColumn}>
-					<Subtitle1>About</Subtitle1>
-					<Body1>
-						{`This Graph shows Fluid Op (Operation) Latency metrics.
-					As you make changes in your Fluid-based application, you'll see this graph update in real time with latency data for any ops your client produces.`}
-						&nbsp;
-						<a
-							target="_blank"
-							rel="noreferrer"
-							href="https://fluidframework.com/docs/concepts/tob/"
-						>
-							{`Learn more about ops.`}
-						</a>
-					</Body1>
-				</div>
-
-				<div className={styles.opPhasesSection}>
-					<Body1Strong>{`Ops in Fluid go through four steps:`}</Body1Strong>
-					<ol>
-						<li>
-							<Body1>Op is submitted by the application.</Body1>
-						</li>
-						<li>
-							<Body1>
-								Op is sent to service. Note: we do not know for sure when the op is
-								actually sent on the network, we only track when it is added to a
-								local outbound queue.
-							</Body1>
-						</li>
-						<li>
-							<Body1>(Sequenced) Op is received from service.</Body1>
-						</li>
-						<li>
+			{renderInstructions ? (
+				<Body1>
+					{`Unsampled telemetry has not been enabled and is necessary for this feature to work as designed. To enable it, open the web console and set the Disable Sampling flag to true using the following command:`}
+					<pre
+						style={{ whiteSpace: "pre-wrap", wordWrap: "break-word" }}
+					>{`localStorage.setItem("Fluid.Telemetry.DisableSampling", "true");`}</pre>
+					{`\nIMPORTANT: This flag is only intended for local development with Devtools and should not be enabled in production scenarios.`}
+				</Body1>
+			) : (
+				<div className={styles.mainContainer}>
```

Comment:
This extra div seems unnecessary? It didn't exist before and the view rendered correctly (I hope 😄).

Reply:
It wasn't necessary before but is needed now because to conditionally render everything else, there needs to be a single parent expression.

Reply:
For that we can use [React Fragments](https://legacy.reactjs.org/docs/fragments.html#short-syntax). `<>` and `</>` can encapsulate everything in a single element without introducing spurious nodes in the rendered DOM.